stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt

```diff
@@ -402,20 +411,30 @@ public class ChannelListViewModel(
     }
 
     private fun loadMoreQueryChannels() {
-        val currentConfig = QueryConfig(
-            filters = filterFlow.value ?: return,
+        logger.d { "[loadMoreQueryChannels] no args" }
+        val currentFilter = filterFlow.value
+        if (currentFilter == null) {
+            logger.v { "[loadMoreQueryChannels] rejected (no current filter)" }
+            return
+        }
+        val currentQuery = queryChannelsState.value?.nextPageRequest?.value
+        if (currentQuery == null) {
+            logger.v { "[loadMoreQueryChannels] rejected (no current query)" }
+            return
+        }
+        val nextQuery = currentQuery.copy(
+            filter = createQueryChannelsFilter(currentFilter, searchQuery.value.query),
             querySort = querySortFlow.value,
         )
-
-        channelsState = channelsState.copy(isLoadingMore = true)
-
-        val currentQuery = queryChannelsState.value?.nextPageRequest?.value
-
-        currentQuery?.copy(
-            filter = createQueryChannelsFilter(currentConfig.filters, searchQuery.value.query),
-            querySort = currentConfig.querySort,
-        )?.let { queryChannelsRequest ->
-            chatClient.queryChannels(queryChannelsRequest).enqueue()
+        viewModelScope.launch {
+            channelsState = channelsState.copy(isLoadingMore = true)
+            val result = chatClient.queryChannels(nextQuery).await()
+            if (result.isSuccess) {
+                logger.v { "[loadMoreQueryChannels] completed; channels.size: ${result.getOrNull()?.size}" }
+            } else {
+                logger.e { "[loadMoreQueryChannels] failed: ${result.errorOrNull()}" }
+            }
+            channelsState = channelsState.copy(isLoadingMore = false)
         }
     }
```

Comment:
Most likely `nextPageRequest` was null, so we never were sending a request to fetch the next page, however, we were setting  `isLoadingMore = true` a line above